R/local_G.R

```diff
@@ -0,0 +1,330 @@
+#' and \deqn{\bar x_i=(\sum_{j=1}^{n}x_j)/(n-1),\forall j\neq i}
+#' The left numerator corresponds to \eqn{G_i}, the right to \eqn{E(G_i)},
+#' and the denominator to \eqn{Var(G_i)}.
+#'
+#' @section \eqn{G_{i}^{*}} ("gstari"):
+#' In the equation for \eqn{G_{i}}, the condition that \eqn{i\neq j} is central. \eqn{G_i^*}
+#' relaxes this constraint, by including \eqn{i} as a neighbor of itself This local measure
+#' is expressed as (Getis and Ord 1996, p. 263, equation 14.3):
+#' \deqn{
+#' Z(G_i^*)=\dfrac{[\sum_{j=1}^{n}w_{i,j}x_j]-[(\sum_{j=1}^{n}w_{i,j})\bar x^*]}
+#' {s^*\{[(n-1)\sum_{j=1}^{n}w_{i,j}^2-(\sum_{j=1}^{n}w_{i,j})^2]/(n-2)\}^{1/2}},\forall j
+#' }
+#' \cr
+#' where \deqn{s^* = \sqrt{((\sum_{j=1}^{n}x_j^2)/n)-\bar x_i^{*2}}}
+#' and \deqn{\bar x_i^*=(\sum_{j=1}^{n}x_j)/n,\forall j}
+#'
+#' @section weights:
+#' Weights are calculated for node pairs if \code{use_weights = TRUE}. For an undirected
+#' path between nodes \eqn{i \sim j}, the weight is defined as the transition probability
+#' for a k-step walk between \eqn{i} and \eqn{j}. This helps normalizing the contribution
+#' by each neighbor in a manner that reduces the influence of high degree nodes. The strategy
+#' rests on the assumption that node degree correlates strongly with the node marker count.
+#'
+#' In practice, the weights are calculated by normalizing the adjacency matrix of the graph
+#' such that each row sums to 1 (also known as the stochastic matrix). By multiplying the
+#' transpose of the stochastic matrix with the count matrix, we obtain a "lag matrix" with the weighted
+#' marker expression for each node neighborhood which is leveraged in the computation of local G.
+#'
+#' When using \code{k > 1}, the k'th power of the stochastic matrix is used instead. In this
+#' case, the transition probabilities are calculated for a k-step random walk.
+#' Since the random walker can reach a larger neighborhood when \code{k > 1}, we will
+#' also include information about the marker expression for nodes that are more than 1 steps
+#' away from the center node. Moreover, when \code{k > 1}, the weights for \eqn{i \sim i}
+#' (self-loops) can become positive. These transitions are not allowed in the calculation of
+#' \eqn{G_i} and are therefore set to 0, and the remaining transition probabilities are normalized
+#' to 1 within each node neighborhood. This weighing scheme is ignored if \code{W} is provided.
+#'
+#' @param g A \code{tbl_graph} object representing an MPX component
+#' @param counts A \code{dgCMatrix} (sparse matrix) with node marker counts. Raw counts
+#' are recommended.
+#' @param W A \code{dgCMatrix} (sparse matrix) with edge weights. If not provided,
+#' the function will use the adjacency matrix of the graph to compute weights.
+#' @param use_weights Logical indicating whether to use edge weights. See section
+#' \strong{weights} for details.
+#' @param normalize_counts Logical indicating whether to normalize the counts. If
```

Comment:
Does this have a big influence on the result? Perhaps remove option if not?

Reply:
It does have a big impact on the results. If activated, the effect will be very similar to using the transition probability weights. 

It might be even better to run `local_G(g, counts, use_weights = FALSE, normalize_counts = TRUE)` than running the default, since we're effectively normalizing each node count vector by the total number of edges (each edge  = 1 counts). However, if we use a `k` larger than 1, we can compute transition probabilities from nodes that are further away which better reflects the distances in the graph. 

These two options: `local_G(g, counts, use_weights = FALSE, normalize_counts = TRUE)` and `localG(g, counts)` should yield fairly similar results.